### ISSUE TITLE
[a11y]: IBM Accessibility checker reports error on Expandable Tile with Interactive

### ISSUE THREAD

**Issue by kglickman** [carbon-design-system/carbon#13759]
### Package

@carbon/react

### Package version

1.23.0

### React version

16.4

### Automated testing tool and ruleset

IBM Equal Accessibility Checker with latest ruleset

### Description

There are 2 issues:

1.  I have a StructuredList inside the Expandable tile.  It doesn't have interactive content so the tile gets rendered as a button.  However, the StructuredList elements have roles and buttons can't have children with roles.

So, to work around this problem, I added a button with display:none in order to force the expandable tile to render the version with interactive content, making the button just the expando icon instead of the whole tile.

Now I run into the issue, reproducible in the storybook, that the tile can't have attribute aria-expanded="false"
![image](https://user-images.githubusercontent.com/30597411/236859794-08cf1fe1-517f-44de-a3ef-febcd6eb0c25.png)

In addition to fixing the aria-expanded issue, it would be great if I could get the "interactive" version without having to put a fake interactive element in my code.

### Reproduction/example

see story book

### Steps to reproduce

see storybook screen shot in description

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by tay1orjones** [comment]
@kglickman Thanks for reporting these. 

> the StructuredList elements have roles and buttons can't have children with roles

Could you share the specific rule you're referring to here? I'm not sure how we should tackle this issue. Right now the check for interactive elements relies on if the elements can receive focus. We could consider expanding the check to assert if child elements have certain roles that would require the ExpandableTile to not be a button. I'd prefer to avoid providing a way to manually toggle the ExpandableTile behavior for interactive vs. not. It would be a huge footgun.


For the second issue when expandable tile has interactive elements, I think the `aria-expands` should be removed from the containing div that has the error. The expando icon `button` should get `aria-controls` with a value of the id of the container (`expandable-tile-1` in the storybook).

**Comment by kglickman** [comment]
The exact error message is "The element with role "button" contains descendants with roles "table" which are ignored by browsers" There is a "Learn more" which says:

This rule is triggered when one of the following elements contain more than presentational descendants: button, checkbox, img, math, menuitemcheckbox, menuitemradio, option, progressbar, radio, scrollbar, separator, slider, switch, and tab. Refer to the table in [ARIA in HTML - Allowed descendants of ARIA roles](https://www.w3.org/TR/html-aria/#allowed-descendants-of-aria-roles) for the normative definition for each ARIA role, the kinds of content categories for each role, and what [kinds of elements](https://html.spec.whatwg.org/multipage/dom.html#kinds-of-content) can be descendants.

For example, a <button> allows text and images, but does not allow [interactive content](https://html.spec.whatwg.org/multipage/dom.html#interactive-content-2) or descendants with a tabindex attribute because that would confuse assistive technology users as to the expected content or behavior of the button.
